auth/src/main/java/com/firebase/ui/auth/ui/phone/PhoneActivity.java

```diff
@@ -14,401 +14,204 @@
-        return createBaseIntent(context, PhoneActivity.class, flowParams)
-                .putExtra(ExtraConstants.PARAMS, params);
+    public static Intent createIntent(Context context, FlowParameters params, Bundle args) {
+        return createBaseIntent(context, PhoneActivity.class, params)
+                .putExtra(ExtraConstants.PARAMS, args);
     }
 
     @Override
     protected void onCreate(@Nullable Bundle savedInstanceState) {
         super.onCreate(savedInstanceState);
         setContentView(R.layout.fui_activity_register_phone);
 
-        mHandler = new Handler();
-        mVerificationState = VerificationState.VERIFICATION_NOT_STARTED;
-        if (savedInstanceState != null && !savedInstanceState.isEmpty()) {
-            mPhoneNumber = savedInstanceState.getString(KEY_VERIFICATION_PHONE);
+        final PhoneProviderResponseHandler handler =
+                ViewModelProviders.of(this).get(PhoneProviderResponseHandler.class);
+        handler.init(getFlowParams());
+        handler.getOperation().observe(this, new ResourceObserver<IdpResponse>(
+                this, R.string.fui_progress_dialog_signing_in) {
+            @Override
+            protected void onSuccess(@NonNull IdpResponse response) {
+                startSaveCredentials(handler.getCurrentUser(), response, null);
+            }
+
+            @Override
+            protected void onFailure(@NonNull Exception e) {
+                handleError(e);
+            }
+        });
 
-            if (savedInstanceState.getSerializable(KEY_STATE) != null) {
-                mVerificationState = (VerificationState) savedInstanceState.getSerializable(
-                        KEY_STATE);
+        final PhoneNumberVerificationHandler phoneVerifier =
+                ViewModelProviders.of(this).get(PhoneNumberVerificationHandler.class);
+        phoneVerifier.init(getFlowParams());
+        phoneVerifier.getOperation().observe(this, new ResourceObserver<PhoneVerification>(
+                this, R.string.fui_verifying) {
+            @Override
+            protected void onSuccess(@NonNull PhoneVerification verification) {
+                if (verification.isAutoVerified()) {
+                    Toast.makeText(
+                            PhoneActivity.this, R.string.fui_verified, Toast.LENGTH_LONG).show();
+                }
+
+                handler.startSignIn(verification.getCredential(), new IdpResponse.Builder(
+                        new User.Builder(PhoneAuthProvider.PROVIDER_ID, null)
+                                .setPhoneNumber(verification.getNumber())
+                                .build())
+                        .build());
             }
-            return;
-        }
+
+            @Override
+            protected void onFailure(@NonNull Exception e) {
+                if (e instanceof PhoneNumberVerificationRequiredException) {
+                    // Ignore if resending verification code
```

Comment:
Can we add some more explanation for how this comment relates to the code below?